pkg/controller/node/node_controller.go

```diff
@@ -381,38 +382,60 @@ func (ctrl *Controller) makeMasterNodeSchedulable(node *corev1.Node) error {
 
 func (ctrl *Controller) addMachineOSBuild(obj interface{}) {
 	curMOSB := obj.(*mcfgv1alpha1.MachineOSBuild)
+	klog.V(4).Infof("Adding MachineOSBuild %s", curMOSB.Name)
 
-	config, _ := ctrl.client.MachineconfigurationV1alpha1().MachineOSConfigs().Get(context.TODO(), curMOSB.Spec.MachineOSConfig.Name, metav1.GetOptions{})
-
+	config, err := ctrl.client.MachineconfigurationV1alpha1().MachineOSConfigs().Get(context.TODO(), curMOSB.Spec.MachineOSConfig.Name, metav1.GetOptions{})
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get MachineOSConfig from MachineOSBuild %#v", curMOSB))
+		return
+	}
+	if config == nil {
```

Comment:
won't be nil unless there was an error.  Is this delete important?  

keep in mind that there is no synchronization between two different caches made by different list/watches.  This is a risky move. 

Can you separate the delete from the panic fix?